devito/operations/interpolators.py

```diff
@@ -268,6 +268,15 @@ def inject(self, field, expr, offset=0):
         offset : int, optional
             Additional offset from the boundary.
         """
+
+        def get_field_offset(field):
+            # In case of nested Functions, use them as a field
+            for ind in field.indices:
+                if ind.is_Function:
```

Comment:
this already fails if `u[..., 1 + f[...]]`

Reply:
should be fixed now, tests added